pdns/dnsdist.hh

```diff
@@ -262,7 +262,75 @@ struct DNSDistStats
   };
 };
 
+// Keeps additional information about metrics
+struct MetricDefinition {
+  MetricDefinition(std::string description, std::string prometheusType) {
+    this->description = description;
+    this->prometheusType = prometheusType;
+  }
+ 
+  MetricDefinition() = default;
+
+  // Metric description
+  std::string description;
+  // Metric type for Prometheus
+  std::string prometheusType;
```

Comment:
Given that we have a limited range of possible values (counter or gauge if I'm not mistaken), perhaps a enum class would be better suited?

Reply:
I reworked it to scoped enum and added function to convert scoped enum value to string representation. Also, I fixed bug about misplaced type and help messages for Prometheus.